csi/volumeutils.py

```diff
@@ -1097,10 +1228,15 @@ def mount_glusterfs(volume, mountpoint, is_client=False):
             "-l", log_file,
             "--volfile-id", volname,
             "--fs-display-name", "kadalu:%s" % volname,
-            "-f", "%s/%s.client.vol" % (VOLFILES_DIR, volname),
+            "-f", client_volfile_path,
```

Comment:
Storage options is being configured and changes are reflected in client volfile for the first request coming in.
After a custom volfile is request is completed, if a default PVC is created with with default SC, error is thrown, since default client volfile is not generated at Line 1121 because of the reason in Line 897.
Can we keep the default or custom client volfile in /tmp or similar?

Reply:
- Didn't review yet, just skimming through issues.
- I feel tmpfile couldn't work for container reboot, is there any possibility to use create tmpfile in empty dir
- But above couldn't work for pod reboot though 😅
- Will get back after reviewing the pr

Reply:
> I feel tmpfile couldn't work for container reboot, is there any possibility to use create tmpfile in empty dir

Persistence not required, only used as base file that gets renamed to actual required file.

> But above couldn't work for pod reboot though

Persistence not required for the generated volfiles, only templates are part of container image.